src/hotspot/share/gc/g1/g1CardSetMemory.cpp

```diff
@@ -79,6 +68,20 @@ void G1CardSetAllocator<Slot>::print(outputStream* os) {
             mem_size());
 }
 
+template <class Slot>
+G1CardSetAllocator<Slot>::G1CardSetAllocator(const char* name,
+                                             const G1CardSetAllocOptions* alloc_options,
```

Comment:
There don't seem to be any specializations of this other than `G1CardSetAllocator<G1CardSetContainer>`.  Unless there are future planned specializations it seems like there's no need for the new non-template base class.  Just get rid of the template parameter and directly replace uses of `Slot` with `G1CardSetContainer`.